cfbs/masterfiles/analyze.py

```diff
@@ -141,5 +144,47 @@ def sort_versions(versions: list, reverse: bool = True):
     )
 
 
-def highest_version(versions):
+def highest_version(versions: Iterable[Version]):
     return max(versions, key=version_as_comparable_list, default=None)
+
+
+def lowest_version(versions: Iterable[Version]):
+    return min(versions, key=version_as_comparable_list, default=None)
+
+
+def is_lower_version(version_a: Version, version_b: Version):
+    """Returns `True` if and only if `version_a` is lower than `version_b`."""
+
+    return version_as_comparable_list(version_a) < version_as_comparable_list(version_b)
+
+
+def most_relevant_version(
+    other_versions: List[Version], reference_version: Version
+) -> Version:
+    """
+    The most relevant version is the highest version among older other versions,
+    or if there are no older other versions, the lowest version among other versions.
+
+    `other_versions` is assumed to be non-empty and not contain `reference_version`."""
+    assert len(other_versions) > 0
+    assert reference_version not in other_versions
+
+    highest_other_version = highest_version(other_versions)
+    lowest_other_version = lowest_version(other_versions)
+    # Unfortunately, Pyright can not infer that these can't be `None` when `other_versions` is non-empty.
+    assert highest_other_version is not None
+    assert lowest_other_version is not None
```

Comment:
These asserts are repeating, maybe it's more useful if we change `highest_version` and `lowest_version` to never return `None`, i.e. to not accept empty iterables? I checked the other places where `highest_version()` is used, and it doesn't seem like a bad change.